src/main/java/hillelee/RestExceptionHandler.java

```diff
@@ -0,0 +1,46 @@
+package hillelee;
+
+import hillelee.doctor.exceptions.DoctorAlreadyExistsException;
+import hillelee.doctor.exceptions.DoctorNotFoundException;
+import hillelee.doctor.exceptions.IdModificationIsNotAllowedException;
+import hillelee.doctor.exceptions.InvalidDoctorSpecialtyException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+import static org.springframework.http.HttpStatus.*;
+
+/**
+ * Created by dmitriy.chebotarev@hpe.com on 12/14/2017.
+ */
+@ControllerAdvice
+public class RestExceptionHandler extends ResponseEntityExceptionHandler
```

Comment:
It is not best idea to have global exception handler.
From my point of view:
* better choice to use exceptions annotated with `@ResponseStatus` (avoid using exception handlers at all)
* or use exception handlers inside controller (where that exceptions a thrown) (avoid using at least global handler)